syntax = "proto3";

service BankService {
    rpc Deposit (DepositRequest) returns (DepositResponse);
    rpc Withdraw (WithdrawRequest) returns (WithdrawResponse);
    rpc Query (QueryRequest) returns (QueryResponse);
}

message DepositRequest {
    int32 customer_id = 1;
    int32 amount = 2;
}

message DepositResponse {
    string result = 1;
}

message WithdrawRequest {
    int32 customer_id = 1;
    int32 amount = 2;
}

message WithdrawResponse {
    string result = 1;
}

message QueryRequest {
    int32 customer_id = 1;
}

message QueryResponse {
    int32 balance = 1;
}
